app/models/batch_process.rb

```diff
@@ -106,6 +106,33 @@ def create_parent_objects_from_oids(oids, metadata_sources, adminset_keys)
       end
     end
   end
+
+  def delete_parent_object(oids, metadata_sources, adminset_keys)
+    admin_set_hash = {}
+    oids.zip(metadata_sources, adminset_keys).each_with_index do |record, index|
+      oid, metadata_source, adminset_key = record
+      admin_set = admin_set_hash[adminset_key]
+      if admin_set.nil?
+        admin_set = AdminSet.find_by_key(adminset_key)
+        admin_set_hash[adminset_key] = admin_set
+      end
+      if admin_set.nil?
+        batch_processing_event("Skipping row [#{index + 2}] with unknown admin set [#{adminset_key}] for parent: #{oid}", 'Skipped Row')
+        next
+      end
+      next unless user_edit_permission(index, admin_set, oid)
+
+      if ParentObject.where(oid: oid).count.negative?
```

Comment:
what does `ParentPbject.destroy` do? is it different from `po = ParentObject.find_by(oid: oid); po.destroy if po.present?`